### PR TITLE
Avoid attaching security groups to NLB in unsupported regions and fix integration test

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
See commits descriptions for details

### Tests
* Tests in the unsupported regions passed

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/constants.py [hunk 1]
```diff
@@ -237,6 +237,7 @@ class Feature(Enum):
     SLURM_DATABASE = "SLURM Database"
     CLUSTER_HEALTH_METRICS = "Cluster Health Metrics"
     CAPACITY_BLOCK = "Capacity Block"
+    NLB_SECURITY_GRROUP = "NLB Security Group"
```
Comment (gmarciani) [inline cli/src/pcluster/constants.py]: 1. typo: GRROUP => GROUP
1. The string is customer facing, so I suggest to expand the acronym NLB to "Network Load Balancer"

File: cli/src/pcluster/constants.py [hunk 2]
```diff
@@ -249,6 +250,7 @@ class Feature(Enum):
     Feature.SLURM_DATABASE: [],
     Feature.CLUSTER_HEALTH_METRICS: ["us-iso"],
     Feature.CAPACITY_BLOCK: ["us-iso"],
+    Feature.NLB_SECURITY_GRROUP: ["us-iso"],
```
Comment (gmarciani) [inline cli/src/pcluster/constants.py]: We should cover this in the unit test `test_Feature_validators.py::test_feature_region_validator` and `test_utils.py::test_is_feature_supported`

**Review by gmarciani — CHANGES_REQUESTED** [review]

File: cli/src/pcluster/constants.py (hunk 1 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/constants.py]: Done

File: cli/src/pcluster/constants.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/constants.py]: Done

**Review by hgreebe — APPROVED** [review]

**Review by gmarciani — APPROVED** [review]

**Pull request merged**